From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Johannes Obermayr <johannesobermayr@gmx.de>
Date: Tue, 2 May 2023 16:58:54 +0200
Subject: [PATCH] 5280: Fix build warnings after 87c16f9

Fixes:
> Task :createJavadocs
/tmp/ghidra/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFDataTypeImporter.java:89: warning: @param argument "importOptions" is not a parameter name.
         * @param importOptions {@link DWARFImportOptions} control optional features during import
           ^
/tmp/ghidra/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFDataTypeManager.java:81: warning: @param argument "builtInDTM" is not a parameter name.
         * @param builtInDTM {@link DataTypeManager} with built-in data types.
           ^
/tmp/ghidra/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFFunction.java:238: warning: @param argument "program" is not a parameter name.
         * @param program Ghidra program that contains the parameter
           ^
/tmp/ghidra/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFFunction.java:254: warning: @param argument "prog" is not a parameter name.
         * @param prog {@link DWARFProgram} that contains this function
           ^
/tmp/ghidra/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFVariable.java:134: warning: @param argument "prog" is not a parameter name.
         * @param prog {@link DWARFProgram}
           ^
/tmp/ghidra/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFVariable.java:149: warning: @param argument "prog" is not a parameter name.
         * @param prog {@link DWARFProgram}
           ^
/tmp/ghidra/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFVariable.java:173: warning: @param argument "prog" is not a parameter name.
         * @param prog {@link DWARFProgram}
           ^
/tmp/ghidra/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/golang/rtti/GoSlice.java:68: warning: @param argument "programContext" is not a parameter name.
         * @param programContext
           ^
8 warnings
---
 .../app/util/bin/format/dwarf4/next/DWARFDataTypeImporter.java | 1 -
 .../app/util/bin/format/dwarf4/next/DWARFDataTypeManager.java  | 1 -
 .../ghidra/app/util/bin/format/dwarf4/next/DWARFFunction.java  | 2 --
 .../ghidra/app/util/bin/format/dwarf4/next/DWARFVariable.java  | 3 ---
 .../java/ghidra/app/util/bin/format/golang/rtti/GoSlice.java   | 1 -
 5 files changed, 8 deletions(-)

diff --git a/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFDataTypeImporter.java b/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFDataTypeImporter.java
index 5cf87925a..93b759a0e 100644
--- a/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFDataTypeImporter.java
+++ b/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFDataTypeImporter.java
@@ -86,7 +86,6 @@ public class DWARFDataTypeImporter {
 	 *
 	 * @param prog {@link DWARFProgram} that is being imported
 	 * @param dwarfDTM {@link DWARFDataTypeManager} helper
-	 * @param importOptions {@link DWARFImportOptions} control optional features during import
 	 */
 	public DWARFDataTypeImporter(DWARFProgram prog, DWARFDataTypeManager dwarfDTM) {
 		this.prog = prog;
diff --git a/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFDataTypeManager.java b/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFDataTypeManager.java
index d3e257c01..0af33bade 100644
--- a/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFDataTypeManager.java
+++ b/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFDataTypeManager.java
@@ -78,7 +78,6 @@ public class DWARFDataTypeManager {
 	 *
 	 * @param prog {@link DWARFProgram} that holds the Ghidra {@link Program} being imported.
 	 * @param dataTypeManager {@link DataTypeManager} of the Ghidra Program.
-	 * @param builtInDTM {@link DataTypeManager} with built-in data types.
 	 * during the import session.
 	 */
 	public DWARFDataTypeManager(DWARFProgram prog, DataTypeManager dataTypeManager) {
diff --git a/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFFunction.java b/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFFunction.java
index 311f5676d..bd7821c3a 100644
--- a/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFFunction.java
+++ b/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFFunction.java
@@ -235,7 +235,6 @@ public class DWARFFunction {
 	 * 
 	 * @param includeStorageDetail boolean flag, if true storage information will be included, if
 	 * false, VariableStorage.UNASSIGNED_STORAGE will be used
-	 * @param program Ghidra program that contains the parameter
 	 * @return list of Parameters
 	 * @throws InvalidInputException
 	 */
@@ -251,7 +250,6 @@ public class DWARFFunction {
 	/**
 	 * Returns a {@link FunctionDefinition} that reflects this function's information.
 	 *  
-	 * @param prog {@link DWARFProgram} that contains this function
 	 * @return {@link FunctionDefinition} that reflects this function's information
 	 */
 	public FunctionDefinition asFuncDef() {
diff --git a/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFVariable.java b/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFVariable.java
index c3f6d14fe..13371400e 100644
--- a/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFVariable.java
+++ b/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/dwarf4/next/DWARFVariable.java
@@ -131,7 +131,6 @@ public class DWARFVariable {
 	 * Assign storage for this variable in a ram data location.
 	 * 
 	 * @param offset address offset
-	 * @param prog {@link DWARFProgram}
 	 */
 	public void setRamStorage(long offset) {
 		clearStorage();
@@ -146,7 +145,6 @@ public class DWARFVariable {
 	 * Assign storage for this variable at a stack offset.
 	 * 
 	 * @param offset stack offset
-	 * @param prog {@link DWARFProgram}
 	 */
 	public void setStackStorage(long offset) {
 		clearStorage();
@@ -170,7 +168,6 @@ public class DWARFVariable {
 	 * Assign storage for this variable via a list of registers.
 	 * 
 	 * @param registers registers that contain the data 
-	 * @param prog {@link DWARFProgram}
 	 */
 	public void setRegisterStorage(List<Register> registers) {
 		clearStorage();
diff --git a/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/golang/rtti/GoSlice.java b/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/golang/rtti/GoSlice.java
index ca3c62479..5317f5558 100644
--- a/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/golang/rtti/GoSlice.java
+++ b/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/golang/rtti/GoSlice.java
@@ -65,7 +65,6 @@ public class GoSlice {
 	 * @param startElement
 	 * @param elementCount
 	 * @param elementSize
-	 * @param programContext
 	 * @return
 	 */
 	public GoSlice getSubSlice(long startElement, long elementCount, long elementSize) {
-- 
2.40.0

